synapse/start.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+if [ ! -f /data/homeserver.yaml ];
+then
+    (
+        cd /data
+        python -m synapse.app.homeserver \
+            --generate-config \
+            --server-name $SERVER_NAME \
+            --keys-directory /data \
+            --generate-keys \
+            --config-path /data/homeserver.yaml \
+            --report-stats=no
+    )
+fi
+exec python -m synapse.app.homeserver --config-path /data/homeserver.yaml
```

Comment:
Maybe adding a `$@` (or whatever actually works, I have to look it up each time) to the exec line so you can add additional arguments if necessary would be good.